### PR TITLE
Initial Support for HTTP/2

### PR DESCRIPTION
**Pull request by graemerocher**
This is an initial PR with support for HTTP/2. 

I have run it through https://github.com/summerwind/h2spec

And it passes and fails on the same tests that raw Netty does.

### CODE REVIEW

File: http-server-netty/src/main/java/io/micronaut/http/server/netty/NettyHttpServer.java [hunk 1]
```diff
@@ -657,4 +604,221 @@ public ChannelGroup getChannelGroup() {
     public WebSocketSessionRepository getWebSocketSessionRepository() {
         return this;
     }
+
+    private HttpToHttp2ConnectionHandler newHttpToHttp2ConnectionHandler() {
+        Http2Connection connection = new DefaultHttp2Connection(true);
+        final InboundHttp2ToHttpAdapter http2ToHttpAdapter = new InboundHttp2ToHttpAdapterBuilder(connection)
+                .validateHttpHeaders(true)
```
Comment (croudet) [inline http-server-netty/src/main/java/io/micronaut/http/server/netty/NettyHttpServer.java]: Don't you need to use `serverConfiguration.isValidateHeaders()` here?

File: http-server-netty/src/main/java/io/micronaut/http/server/netty/NettyHttpServer.java [hunk 2]
```diff
@@ -657,4 +604,221 @@ public ChannelGroup getChannelGroup() {
...
+            Map<String, ChannelHandler> handlers = getHandlerForProtocol(protocol);
+            handlers.forEach(pipeline::addLast);
+        }
+
+        @NotNull
+        private Map<String, ChannelHandler> getHandlerForProtocol(@Nullable String protocol) {
+            final HttpRequestDecoder requestDecoder = new HttpRequestDecoder(NettyHttpServer.this, environment, serverConfiguration);
+            final HttpRequestCertificateHandler requestCertificateHandler = new HttpRequestCertificateHandler();
+            final HttpResponseEncoder responseDecoder = new HttpResponseEncoder(mediaTypeCodecRegistry, serverConfiguration);
+            final RoutingInBoundHandler routingHandler = new RoutingInBoundHandler(
```
Comment (croudet) [inline http-server-netty/src/main/java/io/micronaut/http/server/netty/NettyHttpServer.java]: `HttpRequestCertificateHandler` and `RoutingInBoundHandler` are `@Sharable`. Same instance can be used in all channels.

File: http-server-netty/src/main/java/io/micronaut/http/server/netty/NettyHttpServer.java (hunk 1 above)
Comment (graemerocher) [reply http-server-netty/src/main/java/io/micronaut/http/server/netty/NettyHttpServer.java]: Good catch

File: http-server-netty/src/main/java/io/micronaut/http/server/netty/NettyHttpServer.java (hunk 2 above)
Comment (graemerocher) [reply http-server-netty/src/main/java/io/micronaut/http/server/netty/NettyHttpServer.java]: Will look at this thanks

**Comment by graemerocher** [comment]
Not yet ready for merge, many of the streaming cases are not working

**Pull request merged**